Monika After Story/game/zz_submods.rpy

```diff
@@ -71,6 +72,8 @@ init -1000 python in mas_submod_utils:
         coauthors: constrained_list(constrained_str(regex=LABEL_SAFE_NAME)) = []
         repository: str = ""
         priority: constrained_int(ge=-999, le=999) = 0
+        required_os: frozenset[str] = frozenset()
+        blacklist_os: frozenset[str] = frozenset()
```

Comment:
Can we swap to [this approach](https://docs.pydantic.dev/latest/usage/types/#enums-and-choices) for required/blacklist OS value validation? This should remove the `validate_required_os` func, and simplify the blacklist one, along with enforcing appropriate values only via json schema

This will remove

Reply:
1. The list of possible values isn't static, at least I'm not aware of it. And I'd rather not hardcode values in our code.
2. We'd need at least one of the validators anyway to make sure they have no items in common

Any ideas?

Reply:
Mmm, that brings me another concern actually, seeing as at a glance it's unclear what values are accepted here, I was anticipating a simple windows/mac/linux (to correspond with the `renpy.` checks respectively, does this account for platform versions?

If so do we really want/need to be that specific?

Re: point 2, I'd think we need the second one anyway as it presents some additional logic, so no harm in adding the logic for that there too.

Reply:
I actually was wondering if we want to allow checking versions/architecture, but it might be too much.
I know it will at least return `Linux`, `Darwin`, `Java`, `Windows`. There can be more. Also not sure if it supports android just yet, but at the same time don't want to limit it to only desktops.

>  I was anticipating a simple windows/mac/linux (to correspond with the renpy. checks respectively

Not a fan how renpy handles os checks. However if you think that's sufficient, I can change to that.

Reply:
I think it's best if we base it on what we intend MAS to be supported on, so that means the primary 3 (also Java oh no. Embedded MAS?)
Likewise while we don't support android I see no reason to add OS config for android or iOS.

I don't much like the renpy one myself, however I don't think `platform` is much better either, seeing as it needs to be constantly updated. 
Based on how it acts with the current version in MAS r6, it doesn't even know win10 exists, let alone 11, this is bound to happen again as newer versions of Windows come out too. Therefore esp for versioning or anything else (which may even have a valid use case, but again it might be much for a submod specific thing, and might be better handled within submod logic itself) I don't think platform is a good solution either.

That said, I don't think it's _wrong_ to enforce win/lin/mac, seeing as those are the OSes we support. We have limited support for chromebook and only support them if they run Linux, likewise we don't support android, and I think it's safe to say any offshoot OS is not supported as well because we have zero ability to test for it. Likewsie it's likely people who use niche OSes like those understand this.

Reply:
> also Java oh no. Embedded MAS

Memory hog MAS 👀

> Likewise while we don't support android

I think that can happen eventually (like eventually eventually), so I want to design with that in mind,  the system should be expandable, ideally without adding extra code.

> Based on how it acts with the current version in MAS r6

It's old py2 code, apparently it's different now seeing its sources. So using `platform` is actually the most flexible way.

> That said, I don't think it's wrong to enforce win/lin/mac

In that case, is this list fine: `linux`, `mac`, `windows`? Also, is it `mac` or `darwin`?

Reply:
Oh good, it uses the cmd `ver` command, and not the hardcoded value, okay that's much better so that is reliable.

If android support is a concern we'd need to check what platform returns on droid. As for iOS, I have no idea what it'd be

Reply:
> As for iOS, I have no idea what it'd be

A warning on the screen "get something better"